### ISSUE TITLE
Why does my output say there are only 4 words when there are more in my dictionary

### ISSUE THREAD

**Issue by stevenha29** [cmusphinx/g2p-seq2seq#85]
g2p-seq2seq --train output/output.dic --model g2p-seq2seq-cmudict --max_step 1
Preparing G2P data
Loading vocabularies from g2p-seq2seq-cmudict
Reading development and training data.
Creating model with parameters:
Learning rate:        0.5
LR decay factor:      0.99
Max gradient norm:    5.0
Batch size:           64
Size of layer:        512
Number of layers:     2

Steps per checkpoint: 200
Max steps:            1
Optimizer:            sgd

Reading model parameters from g2p-seq2seq-cmudict
Training done.
Loading vocabularies from g2p-seq2seq-cmudict
Creating 2 layers of 512 units.
Reading model parameters from g2p-seq2seq-cmudict
Beginning calculation word error rate (WER) on test sample.
Words: 4
Errors: 0
WER: 0.000
Accuracy: 1.000
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "BestSplits" device_type: "CPU"') for unknown op: BestSplits
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "CountExtremelyRandomStats" device_type: "CPU"') for unknown op: CountExtremelyRandomStats
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "FinishedNodes" device_type: "CPU"') for unknown op: FinishedNodes
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "GrowTree" device_type: "CPU"') for unknown op: GrowTree
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "ReinterpretStringToFloat" device_type: "CPU"') for unknown op: ReinterpretStringToFloat
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "SampleInputs" device_type: "CPU"') for unknown op: SampleInputs
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "ScatterAddNdim" device_type: "CPU"') for unknown op: ScatterAddNdim
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "TopNInsert" device_type: "CPU"') for unknown op: TopNInsert
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "TopNRemove" device_type: "CPU"') for unknown op: TopNRemove
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "TreePredictions" device_type: "CPU"') for unknown op: TreePredictions
E c:\tf_jenkins\home\workspace\release-win\device\cpu\os\windows\tensorflow\core\framework\op_kernel.cc:943] OpKernel ('op: "UpdateFertileSlots" device_type: "CPU"') for unknown op: UpdateFertileSlots

**Comment by stevenha29** [comment]
This is my dictionary file (I can't attach it for some reason)
arm AA R M
mark M AA R K
eight EY T
torpedo T AO R P IY D OW
launch L AO N CH
fume F Y UW M
whitehead HH W AY T HH EH D
base B EY S
one W AH N
primary P R AY M EH R IY
guidance G IH D AH N S
looks L UH K S
bad B AE D
switching S W IH CH IH NG
to T AH
scs S IY S IY Z
go G OW
active AE K T IH V
on AA N
ecm EH K M
jam JH AE M
em EH M
up AH P
ecm EH K M
active AE K T IH V
switch S W IH CH
from F ER M
the DH AH
secondary S EH K AH N D EH R IY
back B AE K
primary P R AY M EH R IY
control K AH N T R OW L
system S IH S T AH M
pcs P IY K S
mayday M EY D EY
mayday M EY D EY
mayday M EY D EY

**Comment by nshmyrev** [comment]
Tool makes automatic split on test/validation/train subsets. You have 4 words in test subset, the remaining are used for training and validation.

**Comment by stevenha29** [comment]
is there a way to increase the number of words in the test subset or does that number increase as I add more words to my dictionary?

**Comment by nshmyrev** [comment]
You can prepare sets yourself and use --train, --test and --valid. Otherwise test is 10% of the train.

**Comment by stevenha29** [comment]
Can you give me guidance on docs to read up on g2p-seq2seq.  I feel like the README only scratches the surface and I want to make sure I understand what I am doing.  For instance, I do not get why my WER is 0.000.  I know that WER is word error rate but I suspect that there should be some errors.  Am I doing something wrong?
